File: javascript/2. even fibonacci numbers.js
// Even Fibonacci numbers
// https://projecteuler.net/problem=2

function fiboEvenSum(n) {
    let a, b, next, sum;
    a = b = 1;
    next = sum = 0;

    while (true) {
        next = a + b;
        if (next > n) {
            break;
        }
        if (next % 2 == 0) {
            sum += next;
        }
        a = b;
        b = next;
    }

    return sum;
}

console.log(fiboEvenSum(4000000));

File: julia/2. even fibonacci numbers.jl
# Even Fibonacci numbers
# https://projecteuler.net/problem=2

seq = [0, 1]
even_nums = []
while true
    next_num = seq[end] + seq[end-1]
    if next_num > 4000000
        break
    end
    push!(seq, next_num)
    if next_num % 2 == 0
        push!(even_nums, next_num)
    end
end

println(sum(even_nums))
